src/PlayerList.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react";
+import { supabase } from "./supabase_client";
+
+const PlayerList = () => {
+  const [totalPlayers, setTotalPlayers] = useState();
+  const retrievePlayer = async () => {
+    const { data, error } = await supabase
+      .from("Players")
+      .select("name")
+      .eq("roomID", 1234);
+    const playerList = data.map((element) => (
+      <li key={element.name}>{element.name}</li>
+    ));
+    setTotalPlayers(playerList);
```

Comment:
while this is technically ok, you usually want to keep "data" separate from "presentation" and only have data as state variables. html elements like `<li>` are presentation, whereas the player data is data. when you store this variable here, you're storing the `<li>` as well.

If you remember the weird stuff you outputted before (screenshot below), that's what you'll be storing if you store the `<li>`.

![image](https://user-images.githubusercontent.com/101266/171785502-3c3acb03-657c-4df5-987d-f610eaf117fb.png)